service/car_cleaner_service.rb

```diff
@@ -0,0 +1,114 @@
+require 'time'
+require_relative '../model/car_cleaner'
+require_relative '../model/car'
+
+class CarCleanerService
+
+  ONE_HOUR_IN_SECONDS = 3600
+
+  attr_reader :car_cleaner, :time
+
+  def initialize(time)
+    @car_cleaner = CarCleaner.new
+    @time = time
+  end
+
+  def add_car(car)
+    # if the client did not specify the time they need the car, I consider it to be the end of the day
+    car.required_time = Time.parse(time.strftime("%Y-%m-%dT#{today_closing_hour}:00:00%z")) if car.required_time.nil?
+    position_in_queue = -1
+
+    car_cleaner.queue.each do |car_in_queue|
+      next unless car_in_queue.required_time.nil? || car_in_queue.required_time > car.required_time
+      position_in_queue = car_cleaner.queue.index(car_in_queue)
+      break
+    end
+
+    position_in_queue == -1 ? car_cleaner.queue.push(car) : car_cleaner.queue.insert(position_in_queue, car)
+  end
+
+  def simulate_service(number_of_hours)
+    @actions = []
+
+    number_of_hours.times do
+      if is_open?
+        @actions.push("[#{format_time}] Nothing significant happened...") unless handle_car_servicing
+      else
+        @actions.push("[#{format_time}] Car cleaning service is closed until tomorrow morning.")
+      end
+
+      @time += ONE_HOUR_IN_SECONDS
+    end
+
+    handle_car_servicing
+    @actions.push("[#{format_time}] Moved forward in time by #{number_of_hours} hours.")
+  end
+
+  def pick_up_car(car_id)
+    return false unless car_cleaner.cars_ready.has_key?(car_id.to_sym)
+    car_cleaner.cars_ready.delete(car_id.to_sym)
```

Comment:
if we are having a breaking line (like `return`, in this case), it's a good practice to add a newline between it and the next block of the code

Reply:
Makes sense, I will modify them.